test/puppetlabs/trapperkeeper/services_test.clj

```diff
@@ -62,6 +63,55 @@
           app (bootstrap-services-with-empty-config [service1])]
       (is (not (nil? app))))))
 
+(deftest test-restart-file-correct-count
+  (testing "create a restart file and check that it correctly increments when a service starts"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))
+          service2 (service Service1
+                            []
+                            (service1-fn [this] "hi"))]
+          (with-app-with-config app [service1] {:global {:restart-file "./dev-resources/counter"}}
+            (= (slurp "./dev-resources/counter") "1"))
+          (with-app-with-config app [service2] {:global {:restart-file "./dev-resources/counter"}}
+            (= (slurp "./dev-resources/counter") "2"))
+          (fs/delete "./dev-resources/counter"))))
+
+(deftest test-restart-file-HUP-restart
+  (testing "check that restart file correctly increments on HUP restarts"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))]
+      (with-app-with-config app [service1] {:global {:restart-file "./dev-resources/counter"}}
+        (app/restart app)
+        (= (slurp "./dev-resources/counter") "2"))
+      (fs/delete "./dev-resources/counter"))))
+
+(deftest test-restart-file-big-int-limit
+  (testing "restart file should reset to 0 if the counter is too large"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))
+          service2 (service Service1
+                            []
+                            (service1-fn [this] "hi"))]
+      (spit "./bigInt" "9223372036854775807")
+      (with-app-with-config app [service1] {:global {:restart-file "./bigInt"}}
+        (= (slurp "./bigInt") "0"))
+      (with-app-with-config app [service2] {:global {:restart-file "./bigInt"}}
+        (= (slurp "./bigInt") "1"))
+      (fs/delete "./bigInt"))))
+
+(deftest test-invalid-restart-file
+  (testing "restart file should reset to 0 if the file is unparesable"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))]
+      (spit "./invalid" "hello")
+      (with-app-with-config app [service1] {:global {:restart-file "./invalid"}}
+        (= (slurp "./invalid") "0"))
+       (fs/delete "./invalid"))))
+
```

Comment:
Can we add a few tests for things like:

1. Some portion of parent directory structure doesn't exist
2. File is not writable and/or readable